storage/memory/memory.go

```diff
@@ -289,21 +334,21 @@ func (m *memory) Objects(ctx context.Context, s *node.Node, p *predicate.Predica
 	defer close(objs)
 
 	if lo.LatestAnchor {
-		lastTA := make(map[string]*time.Time)
-		trps := make(map[string]*triple.Triple)
-		for _, t := range m.idxSP[spIdx] {
-			p := t.Predicate()
-			ppUUID := p.PartialUUID().String()
-			if p.Type() == predicate.Temporal {
-				ta, err := p.TimeAnchor()
-				if err != nil {
-					return err
-				}
-				if lta := lastTA[ppUUID]; lta == nil || ta.Sub(*lta) > 0 {
-					trps[ppUUID] = t
-					lastTA[ppUUID] = ta
-				}
-			}
+		if lo.FilterOptions != nil {
+			return fmt.Errorf("cannot have LatestAnchor and FilterOptions used at the same time inside lookup options")
+		}
+		lo.FilterOptions = &storage.FilteringOptions{
+			Operation: "latest",
+			Field:     "predicate",
+		}
+		defer func() {
+			lo.FilterOptions = (*storage.FilteringOptions)(nil)
+		}()
```

Comment:
I have just added the comments, but this is only being necessary here because we are not deprecating the bool `LatestAnchor` from `lookupOptions` yet. 

This way, just to keep making the tests related to this bool passing in `memory_test.go` (only place where it is used), I decided the best approach for now was to keep support for it in the volatile OSS driver, but using a `FILTER latest(?p)` behind the drapes as the behavior is the same.
 
About the cleaning of `lo.FilterOptions`, it should not be necessary as I already do this cleaning on the planner level, with `resetFilterOptions` there. But since for `LatestAnchor` I am artificially creating a new `lo.FilterOptions` in the driver level, it is safer to also guarantee that it will be cleaned to `nil` in the driver level too, in the case someone think about adding a test in `memory_test.go` that concatenates a `LatestAnchor` with a normal call to the driver or a `FILTER` for something else there (without calling the planner function `resetFilterOptions` then). This does not happen at the moment (so if I do not add this `defer` there no tests would break) and I doubt someone will do it in the future, but I thought it was safer to add it.

Given that, it brings to light the discussion on **when do we plan to deprecate this `LatestAnchor`** bool from `lookupOptions`. Can we do this in a follow up PR or should we wait a bit more? This could allow us to simplify multiple parts of the code in `memory.go`, deleting all the boilerplate there related to `LatestAnchor`, and also delete a number of tests in `memory_test.go` that at the moment are just retesting `FILTER`.

Reply:
I have talked with Thiago about and it seems that it will be necessary some refactoring in the internal side to deprecate this `LatestAnchor` bool from `lookupOptions`. Then, for now, I will be keeping the support for it in the OSS side as well.